homeassistant/components/vallox/fan.py

```diff
@@ -0,0 +1,170 @@
+"""Support for the Vallox ventilation unit fan."""
+
+import logging
+
+from homeassistant.components.fan import FanEntity
+from homeassistant.core import callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+
+from . import (
+    DOMAIN, METRIC_KEY_MODE, METRIC_KEY_PROFILE_FAN_SPEED_AWAY,
+    METRIC_KEY_PROFILE_FAN_SPEED_BOOST, METRIC_KEY_PROFILE_FAN_SPEED_HOME,
+    SIGNAL_VALLOX_STATE_UPDATE)
+
+_LOGGER = logging.getLogger(__name__)
+
+# Device attributes
+ATTR_PROFILE_FAN_SPEED_HOME = {
+    'description': 'fan_speed_home',
+    'metric_key': METRIC_KEY_PROFILE_FAN_SPEED_HOME
+}
+ATTR_PROFILE_FAN_SPEED_AWAY = {
+    'description': 'fan_speed_away',
+    'metric_key': METRIC_KEY_PROFILE_FAN_SPEED_AWAY
+}
+ATTR_PROFILE_FAN_SPEED_BOOST = {
+    'description': 'fan_speed_boost',
+    'metric_key': METRIC_KEY_PROFILE_FAN_SPEED_BOOST
+}
+
+
+async def async_setup_platform(hass, config, async_add_entities,
+                               discovery_info=None):
+    """Set up the fan device."""
+    if discovery_info is None:
+        return
+
+    client = hass.data[DOMAIN]['client']
+
+    try:
+        await hass.async_add_executor_job(
+            client.set_settable_address, METRIC_KEY_MODE, int)
+
+    except AttributeError as attr_err:
```

Comment:
When can this happen?

Reply:
If wrong arguments would be given: https://github.com/yozik04/vallox_websocket_api/blob/master/vallox_websocket_api/client.py#L163-L178

Could indeed be removed, since it would trigger during local testing. Shall I?

Reply:
We're in control of the arguments, right? Yes, then I think we can remove this try...except.